Use relative aliases for all non-current Grafana aliases

jdbaldry opened the pull request:
Please forgive how overdue this PR is.

Prevents non-latest documentation "stealing" the page away from latest and through permanent redirects for latest pages that no longer exist.

The redirected pages are indexed by search engines but our robots.txt forbids them crawling the non-latest page.

I've checked a bunch of aliases manually by:
1. Running `make docs`
1. Picking an absolute alias path from a random file
1. Entering it in the browser substituting "latest"  for "next" as `make docs` serves at next
1. Checking that I end up at the page represented by the file containing the alias.

I think they are working correctly but I haven't tested them all.

jdbaldry committed:
Use relative aliases for all non-current Grafana aliases

Prevents non-latest documentation "stealing" the page away from latest
and through permanent redirects for latest pages that no longer exist.

The redirected pages are indexed by search engines but our robots.txt
forbids them crawling the non-latest page.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Remove aliases from shared pages

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

robbymilo reviewed (approved):
Looks good. This will also need to be applied to v9.0, (not v9.1) v9.2, and v9.3 as they contain `latest` in the aliases.

jdbaldry committed:
Fix typo in latest alias

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry commented:
> Although looking again there are still some containing `latest` - shouldn't those be `next`?

Good point, this has been addressed in [feca51f](https://github.com/grafana/grafana/pull/60062/commits/feca51fd8d18859d37f1160b04feb85c85cf12c5)

jdbaldry committed:
Remove all current page aliases

find docs/sources -type f -name '*.md' -exec sed -z -i 's#\n *- /docs/grafana/next/[^\n]*\n#\n#' {} \;
find docs/sources -type f -name '*.md' -exec sed -Ez -i 's#\n((aliases:\n *-)|aliases:\n)#\n\2#' {} \;

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Rewrite all current latest aliases to be next

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Prettier

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
